plugin/service/host/plugin.go

```diff
@@ -674,6 +670,27 @@ func awsInstanceToHost(instance types.Instance) (*pb.ListHostsResponseHost, erro
 	return result, nil
 }
 
+// dryRunValidation validates that the IAM role policy attached to the given secrets of the
+// storage bucket has the minimum required permissions needed for this plugin to function
+// as expected. This function will attempt to list AWS hosts.
+func dryRunValidation(ctx context.Context, state *awsCatalogPersistedState, filters ...types.Filter) *status.Status {
```

Comment:
Do we need to pass in a filter? If there are no filters, I believe the function call `DescribeInstances(...)` will return all instances within the region. The reason why I ask is because the dryRunValidation should only be used to validate permission access. I'm worried that a user provides a invalid filter input and causes a bad request error from AWS, which is not what we want to test here.

Testing against the filter should be done outside of this function call.

Reply:
This function here is pulling double-duty. At the moment, we have [checks](https://github.com/hashicorp/boundary-plugin-aws/blob/main/plugin/service/host/plugin.go#L334-L353) in place on `OnCreateSet` and `OnUpdateSet` to describe instances with the filter that is provided when you create the host set. In these two functions, we pass the filter to validate that everything works well.

The `filters` argument is optional, and if you check `OnCreateCatalog` and `OnUpdateCatalog`, there's no filter, we just attempt `DescribeInstances`

Reply:
Thanks for providing me a detailed explanation to help clear things up. I think its a good idea to have a shared function for validating both permissions and the filter value since both code paths will be identical. The only thing that I would ask for is that we improve the function's comment to clarify this behavior because currently the function definition states that it should be used only for checking the minimum required permissions for host catalog functionality.

Reply:
That's a great point - I actually had forgotten to update that description - Done with https://github.com/hashicorp/boundary-plugin-aws/pull/49/commits/247146f8dd7968eb291591919201fc8805269fd1